app/Commands/Sites/CreateCommand.php

```diff
@@ -70,58 +46,76 @@ protected function action(): int
             return self::INVALID;
         }
 
-        $server    = $this->selectServer('deploy to')->first();
-        $userInput = $this->getUserInput();
-        $site      = $this->spinupwp->createSite($server->id, array_merge($this->arguments(), $this->options(), $userInput));
+        $server       = $this->selectServer('deploy to')->first();
+        $this->domain = $this->option('domain') ?? $this->resolveAnswer(['prompt' => 'Domain Name'], $this->nonInteractive());
+        $userInput    = $this->promptForAnswers($this->nonInteractive());
+
+        $site = $this->spinupwp->createSite($server->id, array_merge($this->arguments(), $this->options(), ['domain' => $this->domain], $userInput));
 
         $this->successfulStep("{$site->domain} is {$site->status} (event_id = {$site->eventId()})");
 
         return self::SUCCESS;
     }
 
-    /**
-     * Use different options depending on site install type.
-     */
-    protected function getAvailableOptionIO(string $type): array
+    protected function getPrompts(): array
     {
-        switch ($type) {
+        $prompts = [
+            'https_enabled' => [
+                'default'               => 1,
+                'nonInteractiveDefault' => 0,
+                'type'                  => 'confirm',
+                'prompt'                => 'Enable HTTPS',
+            ],
+            'site_user' => [
+                'prompt'          => 'Site User',
+                'defaultCallback' => [$this, 'getDomainSlug'],
+            ],
+            'db_name' => [
+                'prompt'          => 'Database name',
+                'defaultCallback' => [$this, 'getDomainSlug'],
+            ],
+            'db_pass' => [
+                'prompt'          => 'Database Password',
+                'defaultCallback' => fn () => Str::random(12),
+            ],
+            'wp_title'       => ['prompt' => 'WordPress Title'],
```

Comment:
The single line collapse is cleaner on it's own, but in context I think the inconsistency is contributing to the overall array feeling "messy". 

```php
'wp_title' => [
     'prompt' => 'WordPress Title',
],
```

Reply:
✅